src/main/java/edu/kpi/testcourse/rest/AuthenticatedApiController.java

```diff
@@ -71,4 +77,21 @@ public AuthenticatedApiController(
       );
     }
   }
+
+  @Get(value = "/urls", processes = MediaType.APPLICATION_JSON)
+  public HttpResponse<String> getAll(
+    @Body UrlShortenRequest request,
+    Principal principal,
+    HttpRequest<?> httpRequest,
+    Logic logic) throws UrlRepository.PermissionDenied {
+    try {
+      var aliasList = logic.getAllAliasesForUser(principal.getName());
+      return HttpResponse.ok(json.toJson(aliasList));
+    }
+    catch (PermissionDenied e) {
```

Comment:
It's better to use our lovely UrlRepository interface error
```suggestion
    catch (UrlRepository.PermissionDenied e) {
```